test/unittests/execute_call_test.cpp

```diff
@@ -577,6 +577,77 @@ TEST(execute_call, imported_function_from_another_module_via_host_function)
     EXPECT_THAT(execute(*instance2, 0, {44, 2}), Result(42));
 }
 
+TEST(execute_call, imported_function_with_cotext)
+{
+    /* wat2wasm
+    (import "mod" "foo" (func))
+    (func
+      call 0
+    )
+    */
+    const auto wasm =
+        from_hex("0061736d01000000010401600000020b01036d6f6403666f6f0000030201000a0601040010000b");
+
+    const auto module = parse(wasm);
+
+    constexpr auto host_func = [](std::any& host_context, Instance&, const Value*,
+                                   int) -> ExecutionResult {
+        ++(*std::any_cast<int*>(host_context));
+        return Void;
+    };
+    const auto& host_func_type = module->typesec[0];
+
+    int counter = 0;
+    auto host_context = std::make_any<int*>(&counter);
+
+    auto instance = instantiate(*module, {{{host_func, std::move(host_context)}, host_func_type}});
+
+    EXPECT_THAT(execute(*instance, 0, {}), Result());
+    EXPECT_EQ(counter, 1);
+    EXPECT_THAT(execute(*instance, 1, {}), Result());
+    EXPECT_EQ(counter, 2);
+}
+
+TEST(execute_call, imported_functions_with_shared_cotext)
```

Comment:
Isn't this kind of a duplicate of `resolve_imported_function_duplicate_with_context)`?

Reply:
They are somewhat similar, but in resolve it's a single function that is imported twice and context is copied internally.
Here it's different functions and the user copies the context.